### ISSUE TITLE
'exclude' term of pruning/model_speedup config_list seemingly not working/being recognized

### ISSUE THREAD

**Issue by pmmitche** [microsoft/nni#4815]
**Describe the issue**:
When trying to use the exclude functionality within the config list for model speedups, the exclude term seems to be ignored. I have tried it on my custom model and on a base timm model of efficientnet and experience the same issue. It seems that the exclude does not prevent NNI from trying to prune the layer, leading to the empty layer exception again. I'm wondering if I'm using it wrong, or if this is an issue with the model or a small bug. Note that, for my models, this issue only occurs when the pruning percentage/sparsity starts to get very high. For me, it starts at sparsity=0.8.

Thank you for your time and help so far, and any additional help with this would be greatly appreciated!

**Environment**:
- NNI version: nni==2.7
- Python version: Python 3.8.10
- PyTorch/TensorFlow version: torch==1.10.2+cu113

**Log message**:
The log output from my minimal example can be found here: https://github.com/pmmitche/Masters-Thesis/blob/main/nni_example_output.txt
(Note, I added some print statements in the source code based on the stack trace to see what layer was causing the issue. Those prints are seen in the log).

**How to reproduce it?**:
A minimal working example that highlights the issue (that the log file is based off of) can be found here: https://github.com/pmmitche/Masters-Thesis/blob/main/minimal_pruning_error_example.py

**Comment by zheng-ningxin** [comment]
Hi @pmmitche, thanks for your valuable feedback. The `Exclude` means we do not assign any mask for the target layer, but it may still be affected by the masks of other layers.  For example,
```
x = self.conv1(x)
x = self.conv2(x)
```
If you prune all the output channels of `conv1`, even though you exclude the `conv2` from pruning, it still leads to an empty layer, because the number of the input channels of `conv2` is 0. 
According to you description, I think you just prune the model too hard and some layer is pruned to empty which is not legal in NNI model speedup.
